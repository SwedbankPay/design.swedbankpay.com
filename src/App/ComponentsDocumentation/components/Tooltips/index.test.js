import React from "react";
import { shallow } from "enzyme";

import Tooltip, { Overview } from "./index";

describe("Components: Tooltip", () => {
    it("is defined", () => {
        expect(Tooltip).toBeDefined();
    });

    it("renders", () => {
        const wrapper = shallow(<Tooltip />);

        expect(wrapper).toMatchSnapshot();
    });

    describe("Overview", () => {
        it("is defined", () => {
            expect(Overview).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<Overview />);

            expect(wrapper).toMatchSnapshot();
        });
    });
});
