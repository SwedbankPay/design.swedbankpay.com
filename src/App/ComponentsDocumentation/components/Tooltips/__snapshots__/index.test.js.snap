// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components: Tooltip Overview renders 1`] = `
<Fragment>
  <h2
    id="overview"
  >
    Overview
  </h2>
  <p>
    Add the attribute 
    <CodeTags
      code="tooltip"
      type="secondary"
    />
     to the desired element and add your content in its value to create a tooltip.
  </p>
  <p>
    You can use the attribute 
    <CodeTags
      code="tooltip-position"
      type="secondary"
    />
     to position the tooltip relative to the containing element, use the values 
    <CodeTags
      code="top"
      type="secondary"
    />
    , 
    <CodeTags
      code="right"
      type="secondary"
    />
    , 
    <CodeTags
      code="bottom"
      type="secondary"
    />
     or 
    <CodeTags
      code="left"
      type="secondary"
    />
    . If the attribute is not provided, it will default to top.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    removeOuterTag={true}
    showCasePanel={true}
  >
    <div
      className="button-group"
    >
      

      <button
        className="btn btn-primary"
        data-tooltip="This is a tooltip!"
        data-tooltip-position="left"
        type="button"
      >
        Tooltip to the left
      </button>
      

      <button
        className="btn btn-primary"
        data-tooltip="Payment capture is the process by which payments are secured once the payment has been authorized by the customer."
        data-tooltip-position="top"
        type="button"
      >
        Tooltip on top
      </button>
      

      <button
        className="btn btn-primary"
        data-tooltip="This is a tooltip!"
        data-tooltip-position="bottom"
        type="button"
      >
        Tooltip to the bottom
      </button>
      

      <button
        className="btn btn-primary"
        data-tooltip="Payment capture is the process by which payments are secured once the payment has been authorized by the customer!"
        data-tooltip-position="right"
        type="button"
      >
        Tooltip to the right
      </button>
      

    </div>
  </ComponentPreview>
</Fragment>
`;

exports[`Components: Tooltip renders 1`] = `
<DocContainer
  docToc={true}
>
  <p
    className="lead"
  >
    Tooltips are small, interactive, textual hints for mainly graphical elements. When using icons for actions you can use a tooltip to give people clarification on its function.
  </p>
  <Overview />
  <WithIcons />
  <ExtendedUsage />
  <JavascriptMethods />
</DocContainer>
`;
